src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts

```diff
@@ -689,6 +691,81 @@ class NotebookAccessibilityHelpContribution extends Disposable {
 	}
 }
 
+class NotebookAccessibleViewContribution extends Disposable {
+	static ID: 'chatAccessibleViewContribution';
+	constructor() {
+		super();
+		this._register(AccessibleViewAction.addImplementation(100, 'notebook', accessor => {
+			const accessibleViewService = accessor.get(IAccessibleViewService);
+
+			const activePane = accessor.get(IEditorService).activeEditorPane;
+			const notebookEditor = getNotebookEditorFromEditorPane(activePane);
+			const notebookViewModel = notebookEditor?.getViewModel();
+			const selections = notebookViewModel?.getSelections();
+			notebookViewModel?.getCellIndex;
+			const notebookDocument = notebookViewModel?.notebookDocument;
+
+			if (!selections || !notebookDocument || !notebookEditor?.textModel) {
+				return false;
+			}
+
+			const viewCell = notebookViewModel.viewCells[selections[0].start];
+			let outputContent = '';
+			const decoder = new TextDecoder();
+			for (let i = 0; i < viewCell.outputsViewModels.length; i++) {
+				const outputViewModel = viewCell.outputsViewModels[i];
+				const outputTextModel = viewCell.model.outputs[i];
+				const [mimeTypes, pick] = outputViewModel.resolveMimeTypes(notebookEditor.textModel, undefined);
+				const mimeType = mimeTypes[pick].mimeType;
+				let buffer = outputTextModel.outputs.find(output => output.mime === mimeType);
+
+				if (!buffer || mimeType.startsWith('image')) {
+					buffer = outputTextModel.outputs.find(output => !output.mime.startsWith('image'));
+				}
+
+				let text = `${mimeType}`; // default in case we can't get the text value for some reason.
+				if (buffer) {
+					const charLimit = 100_000;
+					text = decoder.decode(buffer.data.slice(0, charLimit).buffer);
+
+					if (buffer.data.byteLength > charLimit) {
+						text = text + '...(truncated)';
+					}
+
+					if (mimeType.endsWith('error')) {
+						text = text.replace(/\\u001b\[[0-9;]*m/gi, '').replaceAll('\\n', '\n');
+					}
+				}
+
+				const index = viewCell.outputsViewModels.length > 1
+					? `Cell output ${i + 1} of ${viewCell.outputsViewModels.length}\n`
+					: '';
+				outputContent = outputContent.concat(`${index}${text}\n`);
+			}
+
+			if (!outputContent) {
+				return false;
+			}
+
+			accessibleViewService.show({
+				verbositySettingKey: 'notebook',
+				provideContent(): string { return outputContent; },
+				onClose() {
+					notebookEditor?.setFocus(selections[0]);
+					activePane?.focus();
+				},
+				options: {
+					ariaLabel: nls.localize('NotebookCellOutputAccessibleView', "Notebook Cell Output Accessible View"),
+					language: 'plaintext',
+					type: AccessibleViewType.View
+				}
```

Comment:
did you want to add the `next` and `previous` functions here as well? if you pull main, you can

Reply:
I don't see an easy way to synchronously focus on another cell, so I'll leave that out for now.